src/test/java/htsjdk/samtools/cram/structure/SliceTests.java

```
@@ -18,32 +22,121 @@
+                                             final int alignmentStart,
+                                             final int alignmentSpan) {
+        // dummy value
+        final byte[] refBases = null;
+        return dummySliceForTesting(sequenceId, alignmentStart, alignmentSpan, refBases);
+    }
+
+    private static Slice dummySliceForTesting(final int sequenceId,
+                                              final int alignmentStart,
+                                              final int alignmentSpan,
+                                              final byte[] refBases) {
+
+        // dummy values
+
+        final int recordCount = 0;
+        final long globalRecordCounter = 0;
+        final int blockCount = 0;
+        final int[] contentIDs = new int[0];
+        final int embeddedRefBlockContentID = -1;
+        final SAMBinaryTagAndValue tags = null;
+
+        final byte[] refMD5 = SliceHeader.calculateRefMD5(refBases, sequenceId, alignmentStart, alignmentSpan, globalRecordCounter);
+
+        final SliceHeader header = new SliceHeader(sequenceId, alignmentStart, alignmentSpan, recordCount,
+                globalRecordCounter, blockCount, contentIDs, embeddedRefBlockContentID, refMD5, tags);
+
+        return new Slice(header, null, null);
+    }
+
+    private static Slice getUnmappedSlice() {
+        return dummySliceForTesting(SliceHeader.NO_REFERENCE,
+                SliceHeader.NO_ALIGNMENT_START,
+                SliceHeader.NO_ALIGNMENT_SPAN);
+    }
+
     @Test
     public void testUnmappedValidateRef() {
-        Slice slice = new Slice();
-        slice.alignmentStart= SAMRecord.NO_ALIGNMENT_START;
-        slice.sequenceId = SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX;
+        Slice slice = getUnmappedSlice();
 
-        Assert.assertTrue(slice.validateRefMD5(null));
-        Assert.assertTrue(slice.validateRefMD5(new byte[0]));
-        Assert.assertTrue(slice.validateRefMD5(new byte[1024]));
+        slice.validateRefMD5(null);
```

now throws instead of returning boolean